sdk/eventhubs/azure_messaging_eventhubs/src/event_processor/load_balancer.rs

```diff
@@ -0,0 +1,1260 @@
+// Copyright (c) Microsoft Corporation. All Rights reserved
+// Licensed under the MIT license.
+
+// cspell::ignore rngs Seedable
+use super::{
+    models::Ownership,
+    {CheckpointStore, ProcessorStrategy},
+};
+use crate::models::ConsumerClientDetails;
+use azure_core::{error::ErrorKind as AzureErrorKind, Error, Result};
+use rand::{rngs::StdRng, seq::SliceRandom, Rng, RngCore, SeedableRng};
+use std::{
+    cmp::min,
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex, MutexGuard},
+    time::{Duration, SystemTime},
+};
+use tracing::{debug, trace};
+
+/// LoadBalancerInfo contains information about the current ownership of partitions
+/// and the partitions that are unowned or expired.
+/// It also contains information about the maximum number of partitions that can be owned
+/// and whether there are extra partitions that can be claimed.
+/// It is used by the LoadBalancer to determine which partitions to claim ownership of.
+#[derive(Debug)]
+pub struct LoadBalancerInfo {
+    current_ownership: Vec<Ownership>,
+    unowned_or_expired: Vec<Ownership>,
```

Comment:
This sort of stuff doesn't need to be guarded?

Reply:
The `LoadBalancerInfo` struct is never modified once it's returned (we iterate over stuff but we don't modify it). So it doesn't need to be guarded.

I'm *NOT* happy with the number of clones of Ownership that are used here, I am considering making them Arc<Ownership> to make them less heavyweight.